sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "context.hpp"
+#include "http.hpp"
+#include "transport.hpp"
+
+namespace Azure { namespace Core { namespace Http {
+
+    class NextHttpPolicy;
+
+  class HttpPolicy {
+  public:
+    // If we get a response that goes up the stack
+    // Any errors in the pipeline throws an exception
+    // At the top of the pipeline we might want to turn certain responses into exceptions
+    virtual std::unique_ptr<Response> Process(
```

Comment:
Sure.  I used process because so many classes have a Send.  The code isn't easy to read when trying to find references.  It helped the mental model when I was designing this